src/main/java/tree/TargetTreeNode.kt

```diff
@@ -1,9 +1,17 @@
 package tree
 
-interface TargetTreeNode : Comparable<TargetTreeNode> {
-    fun moveRight(): TargetTreeNode
-    fun moveLeft(): TargetTreeNode
-    fun moveUp(): TargetTreeNode
-    fun moveDown(): TargetTreeNode
-    val name: String
+interface TargetTreeNode {
+
+    fun moveToRightSibling(): TargetTreeNode?
+    fun moveToLeftSibling(): TargetTreeNode?
+    fun moveToParent(): TargetTreeNode?
+    fun moveToFirstChild(): TargetTreeNode?
+
+    fun getName(): String
+    fun getTag(): String
+    fun getId(): Int
+    fun setChildren(children: List<TargetTreeNode>)
+    fun getChildren(): List<TargetTreeNode>?
```

Comment:
Do you think 🤔 is better to have an abstract class instead of an interface here. I say it because of the `tag`, `id`, `name` properties. 
In Kotlin there is actually no concept of an abstract class, we can just add properties to the interface, see [here](https://kotlinlang.org/docs/reference/interfaces.html#properties-in-interfaces)

```kotlin
interface TargetTreeNode {
    val name: String
    val tag: String
    val children: List<TargetTreeNode>

    // All method signatures

    fun toString(): String {} // Implementation of toString method.
}
```

@johan-smc Sebs, can you share your opinion in this too please? 🥺

Reply:
I’m not sure, I think more simple is better , but Dorian will chose

Reply:
Well, I tried leaving any trace of implementation out of the interface. But I must admit that I'm not familiar with the best practices in Kotlin, so I could use a little guidance here too.

Reply:
I moved some of the properties from the implementation to the interface, so it should be clearer now.